course6/toyapp_thomasdhuys/drawlib/renderer.cc

```diff
@@ -0,0 +1,99 @@
+#include <iostream>
+
+#include "renderer.h"
+#include "matrix.h"
+
+namespace Draw {
+
+void Renderer::Render(Scenes& scenes)
+{
+	for (auto& scene : scenes) {
+		std::cout << "scene: " << scene << std::endl;
+		Render(scene);
+		window.Display(surface);
+		Display::Delay(5000);
+	}
+}
+
+void Renderer::Render(Scene& scene)
+{
+	for (auto& shape : scene.GetShapes()) {
+		Render(shape);
+	}
+}
+
+void Renderer::Render(Shape& shape)
+{
+	for (auto& primitive : shape.GetPrimitives()){
+
+		if (dynamic_cast<Point*>(primitive.get()))
```

Comment:
Bad design.
If there is really only 3 kinds of primitives, have 3 different methods to call them.
If the universe of primitives is open, then let the primitive write itself.

Reply:
>> If there is really only 3 kinds of primitives, have 3 different methods to call them.
Can you clarify? I'm using 3 different methods here: one Render for points, lines and triangles